src/main/java/seedu/address/model/location/Location.java

```diff
@@ -91,6 +101,29 @@ public boolean isSameId(Location otherLocation) {
                 && otherLocation.getId().equals(getId());
     }
 
+    public static Predicate<Location> getPredicateForHighRiskLocations(Model model) {
+        model.updateFilteredPersonList(PREDICATE_SHOW_ALL_INFECTED);
+        ObservableList<Person> allInfectedPersons = model.getFilteredPersonList();
+
+        HashSet<Index> infectedPersonIds = ModelMethods.getIdHashSetFromPersonsList(allInfectedPersons);
+
+        model.updateFilteredVisitList(Visit.getPredicateForInfectedVisits(infectedPersonIds));
+        ObservableList<Visit> allInfectedVisits = model.getFilteredVisitList();
+
+        ArrayList<Index> infectedLocationIds = ModelMethods.getLocationIdsFromInfectedVisitList(allInfectedVisits);
+
+        model.updateFilteredLocationList(PREDICATE_SHOW_ALL_LOCATIONS);
+        int numberOfTotalLocations = model.getFilteredLocationList().size();
+
+        int numberOfHighRiskLocations = ModelMethods.getNumberOfHighRiskLocations(
+                infectedLocationIds.size(), numberOfTotalLocations);
+
+        ArrayList<Index> highRiskLocationIds =
+                new ArrayList<>(infectedLocationIds.subList(0, numberOfHighRiskLocations));
+
+        return location -> highRiskLocationIds.contains(location.getId());
+    }
+
```

Comment:
Could this be moved to the ModelPredicate class as the predicates are being stored in that class at the moment?

Reply:
Done